src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -12,19 +1,57 @@
+import { FilterByState } from '../../types/FilterByState';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+type Props = {
+  query: string;
+  setQuery: () => void;
+  setFilterBy: () => void;
```

Comment:
The `setFilterBy` prop should be a function that accepts a parameter, likely a string or a specific type related to `FilterByState`, not a function that returns void. Update the type to `(value: string) => void` or the appropriate type.